input/groups/armor.json

```diff
@@ -3047,7 +2973,7 @@
         11534,
         11540
       ],
-      "Standalone: Headgear + Armor Combo": [
+      "Standalone: Headgear + Combo": [
```

Comment:
Yeah, combos have been an absolute headache since forever lmao.

An icon would do wonders, just make it spawn on the bottom right by default and call it a day. Not sure which icons you could use though... Best solution would be a common icon for all combos, then once you click on the item, there's a new section below the trading post price, something like "Combo with:" and the item, so you know what it actually combos with.

Remember, combos are just items that unlock simultaneously with one another, so if you find a way to automate the solution for armor skins that unlock multiple weights at once as well, that would be amazing.

I would really love to nuke the hell out of combo groups once and for all :D.

Reply:
There's no usable source of data on things that unlock together that I'm aware of unfortunately, so we'll have to put something together. An acquisition file would be able to tag, but wouldn't capture what ties together so will need something new for that.

Maybe of the format

```json
[
    {
        "name": "Optional but informational description of the combo, not used", 
        "armor" : [12, 23, 45],
        "weapon": [34]   
    },
    {
        "name": "Another combo",
        "back": [34],
        "glider": [33]
    }
]
```

Reply:
If we're doing something like that I'd like to include secondary skins as well, such as Open Warlord's Plate Helm, Bloodstained Lunatic Noble Coat, Slumbering Obsidian Axe, Guild Cavalier Axe, Ad Infinitum Glider, Musical Minstrel, etc; then remove those from the parent's acquisition method, leaving them under a combo-only filter.

There would be two cases to work with then:

* Items under a package (they get unlocked simultaneously, armor weights can go in here as well).
* Items unlocked through other items (second item unlocked by the first).

I don't think we need to convey which one is the parent item in the second case, as long as the children don't have any non-combo acquisition methods. I understand you're crawling some of them now, might need to change that.

As for the combo name, we can default to the package name itself for the first case, but not sure what to do with the second one, maybe just use the parent item's name and call it a day.

I will create an early "combo.json" file somewhere you feel comfortable with (under input, next to config.json?) then I'll build it for you, won't take me long, I already have most of the data.

I won't include armor weights for the moment, since that one would take over most of the file, and there must be some way to crawl it out of the wiki. If there's not, then I will add them as well.

Reply:
Good point on mutual vs directional tied unlocks. I guess I would split those out a bit. Yeah, stick the combo.json somewhere under input, I can always move it as appropriate.

Wiki crawling typically a bad option, it sort of generally works for vendors but it is always slow and easily broken. It is sort of a choice of last resort. For shared skins for armor weights I think I'ld just do a dump initially of sets of skins with the same icon, as that is usually indicitive (there might be one or two cases where it is wrong though).

Reply:
Cheers. I've actually generated a file with the matching armor weights data, I'll do a bit of work to convert your file to add into that (I ended up with a slightly different format to support non-mutual unlocks). Might end up deleting the matching armor weight entries too, will see how it feels.

I have the site displaying this information locally, still need to do some final tweaks before I'll push it live. I've put clickable links in the details pane that take you to the combo items.

Reply:
I think you can solve non-mutual unlocks by just removing all means of acquisition from the child, only the parent keeping them around. Then both child and parent get the corresponding combo acquisition method, meaning the child will only have the combo as its possible acquisition.

Reply:
That seems to be describing a different issue, which makes me thing we're talking past each other a little. For me non-mutual unlocks is when x unlocks y, but y doesn't unlock x... which might not be a real thing. To me what describing is a just a cleanup of acquisition methods since previously unlocks would be marked with the acquisition method of linked items.

Reply:
The first cut is live now.

Reply:
I don't think there's any case where A unlocks B, but B doesn't unlock A. There are cases where B might not be acquired in any way whatsoever, A being the only way to acquire it. There's also cases where A and B can only be acquired together, never sold separately.

I know there's a few minis and back+glider that are never sold separately, but can still be obtained separately through wardrobe unlockers. You could try to control that, but one, it's not properly documented anywhere at all (not even the wiki), and two, it would be kinda insane to track it up :D.

The philosophy behind combos is to let people know some skins are never sold separately, just that. Whether unlocks apply or not, that's a whole different issue (I don't think wardrobe unlockers have repeated entries for the same unlock set anyway). Combos should only be covering restricted packages and skin variants, nothing more.

As I view it, if you're going to spend wardrobe unlockers and you're missing 5 miniatures that come packed together, you might want to buy said package, because that's 5 wardrobe unlockers you'll be wasting otherwise. That's the information this whole combo thing should be conveying, in my opinion :D.